app/reducers/optional.js

```diff
@@ -1,7 +0,0 @@
-//this is a simple pass thru that offers an extension point
```

Comment:
So just to be clear, here _was_ my understanding. Given a state tree like so:

![image](https://cloud.githubusercontent.com/assets/980757/18566232/f812999a-7b8a-11e6-8fb4-27e032916f79.png)

Your `reducers/index.js` would be:

````js
import { combineReducers } from 'redux';
import auth from './auth';
import identity from './identity';
import presence from './presence';
import messages from './messages';
import outstanding from './outstanding';

export default combineReducers({
  auth,
  identity,
  presence,
  messages,
  outstanding
});
````

This in turn would make the `service/redux.js` (commented out lines are changes):

````js
import Ember from 'ember';
import redux from 'npm:redux';
import reducers from '../reducers/index';
import enhancers from '../enhancers/index';
// import optional from '../reducers/optional';
import initialState from '../reducers/initialState';
import middleware from '../middleware/index';

var { createStore, applyMiddleware, combineReducers, compose } = redux;
var createStoreWithMiddleware = compose(applyMiddleware(...middleware), enhancers)(createStore);

export default Ember.Service.extend({
    init() {
        // this.store = createStoreWithMiddleware(optional(combineReducers(reducers)));
        this.store = createStoreWithMiddleware(reducers, initialState);
        this._super(...arguments);
    },
    getState() {
        return this.store.getState();
    },
    dispatch(action) {
        return this.store.dispatch(action);
    },
    subscribe(func) {
        return this.store.subscribe(func);
    }
});
````